http/src/main/java/io/micronaut/http/body/TypedMessageBodyReader.java

```diff
@@ -38,6 +38,6 @@ public interface TypedMessageBodyReader<T> extends MessageBodyReader<T> {
 
     @Override
     default boolean isReadable(Argument<T> type, MediaType mediaType) {
-        return type.isAssignableFrom(getType());
+        return type.isAssignableFrom(getType()) && !type.getType().equals(Object.class);
```

Comment:
I don't think this a correct fix, typed reader should only add a type and nothing else

Reply:
I'm open to suggestions. The new test cases cover the issue pretty well.